runtime/shared_common/CacheMap.cpp

```diff
@@ -841,12 +1009,12 @@ SH_CacheMap::initializeROMSegmentList(J9VMThread* currentThread)
 	Trc_SHR_Assert_True(_sharedClassConfig != NULL);
 	Trc_SHR_CM_initializeROMSegmentList_Entry(currentThread);
 
-	cacheBase = (U_8*)_cc->getBaseAddress();
```

Comment:
The segments need to be loaded for all the caches.

Reply:
Fixed. I added a new field metadataMemorySegment to J9SharedClassCacheDescriptor. I still let J9SharedClassConfig->metadataMemorySegment point to the metadata segment of _ccHead. The only place I find J9SharedClassConfig->metadataMemorySegment being used is shcReadSharedCacheMetadata(). But the APIs in shcdatautils.c are not being used anywhere. The comment says 
"This is designed to match the dbg version of this function which is in dbgext.c"

I don't see dbgext.c. I guess it is dead code. The field J9SharedClassConfig->metadataMemorySegment and the APIs in shcdatautils.c can be eventually removed.